core/src/main/scala/kafka/admin/TopicCommand.scala

```diff
@@ -271,6 +275,49 @@ object TopicCommand extends Logging {
     ret.toMap
   }
 
+  private[admin] class PartitionsDescription(topic: String, partitions: List[(Int, Seq[Int])], liveBrokers: Set[Int], topicAndPartitionToLeaderAndIsr: Map[(String, Int), (Option[Int], Seq[Int])]) {
+
+    def describeAndPrint(): String = {
+      val sb = new StringBuilder
+      for ((partitionId, assignedReplicas) <- partitions) {
+        val (leader, inSyncReplicas) = topicAndPartitionToLeaderAndIsr.get(topic, partitionId).get
+        if (shouldPrint(assignedReplicas, leader, inSyncReplicas)) {
+          sb.append(printIt(topic, partitionId, assignedReplicas, leader, inSyncReplicas))
+        }
+      }
+      sb.toString()
+    }
+
+    def shouldPrint(assignedReplicas: scala.Seq[Int], leader: Option[Int], inSyncReplicas: scala.Seq[Int]): Boolean = true
+
+    def printIt(topic: String, partitionId: Int, assignedReplicas: scala.Seq[Int], leader: Option[Int], inSyncReplicas: scala.Seq[Int]): String = {
```

Comment:
It is pretty confusing to have both describeAndPrint  and printIt methods. Since describeAndPrint has at least parts of the printing logic, I think it makes sense to merge them.